### PR TITLE
[Vue] remove angularjs libraries from matomo

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

**This PR is based off of #19496.**

Changes:
* Remove angularjs libs from git and from inclusion in CoreHome.php/other controllers.
* Some code changes to get everything to build w/o angularjs.

image

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by peterhashair** [comment]
@diosmosis this PR is stated for a while, but when I checked this PR, somewhere got else tag error.

**Comment by diosmosis** [comment]
@peterhashair #19496 should be reviewed and merged first

**Review by sgiehl — CHANGES_REQUESTED** [review]
I've merged in the latest changes and applied a couple of fixes.
In general this looks good now.
@diosmosis The AngularJS tests are now failing on travis. Could you check if we can simply remove that? Not sure if it still executes some other tests...

**Review by sgiehl — CHANGES_REQUESTED** [review]
clicked through the UI for a while and wasn't able to find any issues in core or with the submodule plugins. UI tests also seem to pass.
I've searched the code for occurrences of `angular` and there are still a lot matches. Could you check those and remove it where no longer needed, rename variables, ...
Also it seems this file was left over and can be removed: https://github.com/matomo-org/matomo/blob/40afe8b101537e1ca4c28d345b756a22cbaeac21/plugins/CorePluginsAdmin/vue/src/Form/Form.adapter.ts

**Comment by diosmosis** [comment]
@sgiehl updated. requires a travis-scripts pr to be merged too

File: core/Common.php [hunk 1]
```diff
@@ -580,38 +580,6 @@ public static function getRequestVar($varName, $varDefault = null, $varType = nu
...
 
-    /**
-     * Replaces lbrace with an encoded entity to prevent angular from parsing the content
-     *
-     * @deprecated Will be removed, once the vue js migration is done
-     *
-     * @param $string
-     * @return array|string|string[]|null
-     */
-    public static function fixLbrace($string)
```
Comment (sgiehl) [inline core/Common.php:591]: We should mention in the changelog that this method has been removed.

**Review by sgiehl — CHANGES_REQUESTED** [review]
Looks good so far. Let's see if the tests are passing.
The travis PR might need another change though.
Last thing we need to do is to mention the removal of angularjs in the changelog.

**Comment by diosmosis** [comment]
@sgiehl updated

**Pull request merged**